src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  for (let i = 0; i < actions.length; i++) {
+    if (actions[i].type === 'addProperties') {
+      currentState = { ...currentState, ...actions[i].extraData };
+
+      result.push(currentState);
+    }
+
+    if (actions[i].type === 'removeProperties') {
+      const newState = { ...currentState };
+
+      for (const property of actions[i].keysToRemove) {
+        delete newState[property];
+      }
+      currentState = newState;
+      result.push(currentState);
+    }
+
+    if (actions[i].type === 'clear') {
+      currentState = {};
```

Comment:
Clearing the state by assigning an empty object is correct. Ensure that this behavior is intended, as it completely resets the state.